osgeo_importer/tests.py

```diff
@@ -147,6 +147,57 @@ def generic_import(self, file, configuration_options=[{'index': 0}]):
 
         return layer_results[0]
 
+    def generic_api_upload(self, files, configuration_options=None):
+        """
+        Tests the import api.
+        """
+        c = AdminClient()
+        c.login_as_non_admin()
+        
+        # Upload Files
+        if isinstance(files, type(str())):
+            files = [files]
+        outfiles = []
+        handles = {}
+        for file in files:
+            f = os.path.join(
+                os.path.dirname(__file__),
+                '..',
+                'importer-test-files',
+                file)
+            handles[file] = open(f)
+            outfiles.append(SimpleUploadedFile(file, handles[file].read()))
+        response = c.post(
+            reverse('uploads-new-json'),
+            {'file': outfiles,
+             'json': json.dumps(configuration_options)},
+            follow=True)
+        # Clean up file handles
+        for file, handle in handles.items():
+            handle.close()
+        content = json.loads(response.content)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(content['id'], 1)
+
+        # Configure Uploaded Files
+        uploadid = content['id']
+        upload_layers = UploadLayer.objects.filter(upload_id=uploadid)
+        
+        for upload_layer in upload_layers:
+            print upload_layer.id
+            print upload_layer.name
+            for config in configuration_options:
+                if config['upload_file_name'] == upload_layer.name:
+                    payload = config['config']
+                    response = c.post('/importer-api/data-layers/{0}/configure/'.format(upload_layer.id), data=json.dumps(payload),
```

Comment:
more readable and no line length issue if url is constructed, then passed to c.post